packages/cli/lib/base-generator.js

```diff
@@ -297,8 +296,48 @@ module.exports = class BaseGenerator extends Generator {
           'The command must be run in a LoopBack project.',
       );
       this.exit(err);
+      return;
     }
     this.packageJson = pkg;
+
+    const projectDeps = pkg.dependencies || {};
+    const projectDevDeps = pkg.devDependencies || {};
+
+    const cliPkg = require('../package.json');
+    const templateDeps = cliPkg.config.templateDependencies;
+    const incompatibleDeps = {};
+    for (const d in templateDeps) {
+      const versionRange = projectDeps[d] || projectDevDeps[d];
+      if (!versionRange || semver.intersects(versionRange, templateDeps[d]))
+        continue;
+      incompatibleDeps[d] = [versionRange, templateDeps[d]];
+    }
+
+    if (Object.keys(incompatibleDeps).length === 0) {
+      // No incompatible dependencies
+      return;
+    }
+    this.log(
+      chalk.red(
+        'The project has dependencies with incompatible versions required by the CLI:',
+      ),
+    );
+    for (const d in incompatibleDeps) {
+      this.log(chalk.yellow('- %s: %s (cli %s)'), d, ...incompatibleDeps[d]);
+    }
+    const prompts = [
+      {
+        name: 'ignoreIncompatibleDependencies',
+        message: `Continue to run the command?`,
+        type: 'confirm',
+        default: false,
+      },
+    ];
+    const answers = await this.prompt(prompts);
+    if (answers && answers.ignoreIncompatibleDependencies) {
```

Comment:
Will this work in non-interactive mode when the answers are provided via JSON and optional prompts are skipped?

Reply:
yes